core/unix/os_public.h

```diff
@@ -42,6 +42,10 @@
 #ifndef _OS_PUBLIC_H_
 #define _OS_PUBLIC_H_ 1
 
+#ifdef LINUX
+# include "include/sigcontext.h"
```

Comment:
I think the include in signal.c is needed, but the only signal-ish thing exported is sigcontext_t (and it is weird how signal.h would have been used to do that when the whole sigcontext header tried to avoid such a thing)